src/app/db/tasks.py

```diff
@@ -0,0 +1,22 @@
+from databases import Database
+from fastapi import FastAPI
+
+from app.core.logging import logger
+from app.db.core import database_url
+
+
+async def connect_to_db(app: FastAPI) -> None:
+    database = Database(database_url(), min_size=2, max_size=10)
+
+    try:
+        await database.connect()
+        app.state._db = database
```

Comment:
Not exactly sure how/what the database package does, but if it does not internally maintain some sort of connection pool, what could happen is that postgres closes the connection after a couple of minutes of inactivity and then the app would have a dangling `state._db` object that can't be used.

Reply:
That is indeed a very good point. BUT: database is a context manager and I presume/hope/pray that calling it with FastAPI creates such a context manager. This means, if the call is over ,the connection is closed. I added disconnects to all other connections, hoping this makes it cleaner

Reply:
I found no immediate solution to enable the .connection as a context manager - I would have to supply the direct database object itself - not initialized ere. Could be an option tot, but would require significant refactoring

Reply:
Yeah, I would say we leave that for the future, its just something we shouldn't be to surprised about if we encounter some problems :-)